service/timeline.go

```diff
@@ -93,10 +93,16 @@ func (uis *UIServer) patchTimelineWrapper(author string, w http.ResponseWriter,
 
 func (uis *UIServer) patchTimelineJson(w http.ResponseWriter, r *http.Request) {
 	projCtx := MustHaveProjectContext(r)
+	projRef, err := projCtx.GetProjectRef()
```

Comment:
I think it's possible to get the project_id out of the request with something like 
```
	vars := gimlet.GetVars(r)
	projectId := vars["project_id"]
```

Also, I think the case where `projCtx.GetProject()` returns a nil project might be when `projCtx.GetProjectRef()` itself returns nil
https://github.com/evergreen-ci/evergreen/blob/b58f3cf6aa534dbccce25922f3bd0d0acd28c20f/model/context.go#L77

Reply:
Although the `project_id` variable won't be there in the case where the request is for the user's project, we only actually **need** it here
 https://github.com/evergreen-ci/evergreen/blob/b58f3cf6aa534dbccce25922f3bd0d0acd28c20f/service/timeline.go#L117
and in that case we will have it.

Reply:
patchTimeLineJson handles one route where the project_id isn't part of the request -- that leads me to think we can't rely on getting it from the request variables?

https://github.com/evergreen-ci/evergreen/blob/b58f3cf6aa534dbccce25922f3bd0d0acd28c20f/service/ui.go#L251

Reply:
Didn't see! Pushed changes that get the project identifier from the request only when we need it.